src/index.ts

```diff
@@ -53,4 +55,149 @@ const launchServer = async () => {
+        'https://alimentaide-973-guyane.oortcloud.tech/api';
+
+      // Check if the server is GUYANE
+      if (!IS_ALIMENTAIDE) {
+        return;
+      }
+
+      // Get all users with lastLogin 6 months ago
+      const usersToDelete = await User.find({
+        lastLogin: { $lt: new Date(Date.now() - 6 * 30 * 24 * 60 * 60 * 1000) }, // 6 months ago
+      });
+
+      // Find the resource with name staff
+      const staff = await Resource.findOne({ name: 'Staff' });
+
+      // Find all the records of staff with the users found above
+      const usersStaffRecords = await Record.find({
+        resource: staff._id,
+        user: { $in: usersToDelete.map((user) => user._id) },
+      });
+
+      // Hiding the info both on the user and their staff record
+      // username/email:  [RANDOM STRING]@anonymus-oort.com
+      // name: [ANONYMOUS]
+
+      // Hide the info on the user
+      for (let i = 0; i < usersToDelete.length; i++) {
+        const user = usersToDelete[i];
+
+        // Update the user
+        user.username = `${Math.random()
+          .toString(36)
+          .substring(2, 15)}@anonymus-oort.com`;
+        user.firstName = 'ANONYMOUS';
+        user.lastName = 'ANONYMOUS';
+        user.name = 'ANONYMOUS';
+        user.roles = [];
+
+        await user.save();
+      }
+
+      // Hide the info on the staff record
+      for (let i = 0; i < usersStaffRecords.length; i++) {
+        const staffRecord = usersStaffRecords[i];
+
+        // Update the staff record
+        staffRecord._createdBy = new User({
+          name: 'ANONYMOUS',
+          username: `${Math.random()
+            .toString(36)
+            .substring(2, 15)}@anonymus-oort.com`,
+        });
+        staffRecord.data.nom_employes = 'ANONYMOUS';
+        staffRecord.data.prenom_employes = 'ANONYMOUS';
+        staffRecord.data.nom_prenom_employes = 'ANONYMOUS';
+        staffRecord.data.tel_staff = 'ANONYMOUS';
+        staffRecord.data.email_staff = 'ANONYMOUS';
+        staffRecord.data.birthdate_employes = 'ANONYMOUS';
+        staffRecord._lastUpdatedBy = new User({
+          name: 'ANONYMOUS',
+          username: `${Math.random()
+            .toString(36)
+            .substring(2, 15)}@anonymus-oort.com`,
+        });
+        staffRecord.data.file_gdpr_staff = [];
+
+        await staffRecord.save();
+      }
+
+      // For the beneficiaries
+      // Trigger when: The last aid to the family was given more than 18 months ago
+      // Than: Anonymize all members of that family
+
+      // Find the resource with name Aid
+      const Aid = await Resource.findOne({ name: 'Aid' });
+
+      // Find all the records of Ais was given more than 18 months ago
+      const AidRecords = await Record.find({
+        resource: Aid._id,
+        createdAt: {
+          $lt: new Date(Date.now() - 18 * 30 * 24 * 60 * 60 * 1000),
+        }, // 18 months ago
+      });
+
+      // Anonymize all members of that family
```

Comment:
This logic is incorrect, like this it would erase all beneficiaries that got an aid longer than 18 months ago, when in fact we want to erase those who has *NOT* gotten an aid in the last 18 months.

Reply:
I added a new commit to fix that!